src/integration/java/io/pinecone/integration/dataplane/UpdateAndQueryTest.java

```diff
@@ -0,0 +1,438 @@
+                .setId(upsertIds.get(0))
+                .setNamespace(namespace)
+                .addAllValues(updatedValues)
+                .build();
+
+        blockingStub.update(updateRequest);
+
+        // Query by ID to verify
+        QueryRequest queryRequest = QueryRequest.newBuilder()
+                .setId(upsertIds.get(0))
+                .setNamespace(namespace)
+                .setTopK(1)
+                .setIncludeValues(true)
+                .build();
+        QueryResponse queryResponse = blockingStub.query(queryRequest);
+        List<Float> queryResponseValues = queryResponse.getMatches(0).getValuesList();
+
+        assertEquals(updatedValues.size(), queryResponseValues.size());
+        int expectedValueSum = 0, actualValueSum = 0;
+        for (int i = 0; i < updatedValues.size(); i++) {
+            expectedValueSum += updatedValues.get(i);
+            actualValueSum += queryResponseValues.get(i);
+        }
+        assertEquals(expectedValueSum, actualValueSum);
+    }
+
+    @Test
+    public void UpdateAllParamsFetchAndQuerySync() throws InterruptedException {
+        // Upsert vectors with required parameters
+        List<String> upsertIds = Arrays.asList("v1", "v2", "v3");
+        String expectedId = upsertIds.get(0);
+        String namespace = RandomStringBuilder.build("ns", 8);
+        blockingStub.upsert(buildRequiredUpsertRequest(upsertIds, namespace));
+        Thread.sleep(3500);
+
+        // Verify the upserted vector count with fetch
+        FetchRequest fetchRequest = FetchRequest.newBuilder().addAllIds(upsertIds).setNamespace(namespace).build();
+        FetchResponse fetchResponse = blockingStub.fetch(fetchRequest);
+        assertEquals(fetchResponse.getVectorsCount(), upsertIds.size());
+        for (String key : upsertIds) {
+            assert (fetchResponse.containsVectors(key));
+        }
+
+        List<Float> updatedValues = Arrays.asList(101F, 102F, 103F);
+        HashMap<String, List<String>> metadataMap = createAndGetMetadataMap();
+        Struct metadata = Struct.newBuilder()
+                .putFields(metadataFields[0],
+                        Value.newBuilder().setStringValue(metadataMap.get(metadataFields[0]).get(0)).build())
+                .putFields(metadataFields[1],
+                        Value.newBuilder().setStringValue(metadataMap.get(metadataFields[1]).get(0)).build())
+                .build();
+
+        SparseValues sparseVector = SparseValues
+                .newBuilder()
+                .addAllIndices(sparseIndices)
+                .addAllValues(sparseValues)
+                .build();
+
+        // Update required fields only
+        UpdateRequest updateRequest = UpdateRequest.newBuilder()
+                .setId(expectedId)
+                .setNamespace(namespace)
+                .addValues(updatedValues.get(0))
+                .addValues(updatedValues.get(1))
+                .addValues(updatedValues.get(2))
```

Comment:
Do we need to explicitly add each value from the `List` via `.get(0)` etc? Can we just do `.addAllValues(updatedValues)`?

Reply:
We can do addAllValues but I'd like to test this functionality here. addAllValues has been tested in a different test. Our goal is to test as many functionalities as possible and randomized it as well at the same time we can to mimic the real world scenario.